src/librustc_trait_selection/traits/mod.rs

```diff
@@ -136,7 +136,7 @@ pub fn type_known_to_meet_bound_modulo_regions<'a, 'tcx>(
     let trait_ref = ty::TraitRef { def_id, substs: infcx.tcx.mk_substs_trait(ty, &[]) };
     let obligation = Obligation {
         param_env,
-        cause: ObligationCause::misc(span, hir::DUMMY_HIR_ID),
+        cause: ObligationCause::misc(span, hir::CRATE_HIR_ID),
```

Comment:
cc @nikomatsakis Is this a good way to encode this?

Reply:
Oh this is what `ObligationCause::dummy` does, heh.

Reply:
Yes, i just "copied" what was done in `dummy`